packages/core/src/lib/filter/index.ts

```diff
@@ -1,260 +1,38 @@
-import { Stream } from "@libp2p/interface/connection";
-import type { Peer } from "@libp2p/interface/peer-store";
 import type { IncomingStreamData } from "@libp2p/interface-internal/registrar";
 import type {
-  Callback,
-  ContentTopic,
   IAsyncIterator,
   IDecodedMessage,
   IDecoder,
   IFilter,
-  IProtoMessage,
   IReceiver,
   Libp2p,
   PeerIdStr,
   ProtocolCreateOptions,
   PubSubTopic,
   Unsubscribe
 } from "@waku/interfaces";
-import { WakuMessage } from "@waku/proto";
 import {
   ensurePubsubTopicIsConfigured,
   groupByContentTopic,
   toAsyncIterator
 } from "@waku/utils";
 import { Logger } from "@waku/utils";
-import all from "it-all";
 import * as lp from "it-length-prefixed";
 import { pipe } from "it-pipe";
 
 import { BaseProtocol } from "../base_protocol.js";
 import { DefaultPubSubTopic } from "../constants.js";
 
-import {
-  FilterPushRpc,
-  FilterSubscribeResponse,
-  FilterSubscribeRpc
-} from "./filter_rpc.js";
-
-const log = new Logger("filter:v2");
+import { FilterPushRpc } from "./filter_rpc.js";
+import { Subscription } from "./subscription.js";
```

Comment:
Maybe `SubscriptionManager`? `Subscription` is a good name but so simple it could be easy to misunderstand.

Reply:
as it is for internal usage I guess it's not that important